at_client/lib/src/client/at_client_impl.dart

```diff
@@ -107,26 +108,27 @@ class AtClientImpl implements AtClient {
       String _atSign, String? namespace, AtClientPreference preference) {
     currentAtSign = AtUtils.formatAtSign(_atSign);
     _preference = preference;
+    _preference.namespace = namespace;
     _namespace = namespace;
   }
 
   Future<void> _init() async {
-    if (_preference!.isLocalStoreRequired) {
+    if (_preference.isLocalStoreRequired) {
       _localSecondary = LocalSecondary(this);
     }
```

Comment:
Need to be really clear if LocalSecondary can be null or not. Right now given this code, it can still be null after initialization - but the declaration above has changed from `LocalSecondary? _localSecondary;` to `late LocalSecondary _localSecondary;` and as a result code which accesses the localSecondary is no longer required to do null checks, although that code should still be doing null checks.

Reply:
Removed the check, _preference.isLocalStoreRequired. LocalSecondary gets initialized by default and added depricated tag for isLocalStoreRequired in AtClientPreferences